File: app/src/main/java/com/zhufuc/pctope/ActivityCollector.java
package com.zhufuc.pctope;

import android.app.Activity;

import java.util.ArrayList;
import java.util.List;

/**
 * Created by zhufu on 17-6-23.
 */

public class ActivityCollector {
    public static List<Activity> activities = new ArrayList<>();
    public static void addActivity(Activity activity){
        activities.add(activity);
    }
    public static void removeActivity(Activity activity){
        activities.remove(activity);
    }
    public static void finishActivity(Activity activity){
        activity.finish();
    }
    public static void finishAll(){
        for (Activity activity : activities){
            if(!activity.isFinishing()){
                activity.finish();
            }
        }
    }
}

File: app/src/main/java/com/zhufucdev/pctope/collectors/ActivityCollector.kt
package com.zhufucdev.pctope.collectors

import android.app.Activity
import com.zhufucdev.pctope.utils.mLog

import java.util.ArrayList

/**
 * Created by zhufu on 17-6-23.
 */

object ActivityCollector {
    private val activities = ArrayList<Activity>()
    fun addActivity(activity: Activity) {
        activities.add(activity)
        mLog.d("Activity", "Added " + activity.javaClass.simpleName)
    }

    fun removeActivity(activity: Activity) {
        activities.remove(activity)
        mLog.d("Activity", "Removed " + activity.javaClass.simpleName)
    }

    fun finishOther(activity: Activity) {
        for (act in activities) {
            if (act != activity) {
                if (!act.isFinishing) {
                    act.finish()
                }
            }
        }
    }

    fun finishAll() {
        for (activity in activities) {
            if (!activity.isFinishing) {
                activity.finish()
            }
        }
    }
}
